WooCommerce/Classes/ViewRelated/Orders/Order Details/Order Summary Section/Edit Order Status/OrderStatusListViewController.swift

```diff
@@ -36,8 +36,9 @@ final class OrderStatusListViewController: UIViewController {
     ///
     private let order: Order
 
-    init(order: Order) {
+    init(order: Order, currentStatus: OrderStatus?) {
```

Comment:
I'm just curious, but have you considered passing the view model? (considering both order and status are provided by it)

Reply:
the view model has both the `Order` and the `OrderStatus?`, but also a few other public properties and functions (like `products: [Product]`, `trackingIsReachable` etc). I thought it might look more clear with just the necessary models for the init with minimum dependencies. Did you have other thoughts?

Reply:
Fair enough. I prefer to pass objects more than single properties, though, for two reasons: first, we are leaking the abstraction (the code that instantiates OrderStatusListViewController needs to know the exact properties of the view model that the view controller needs in order to do its job, and the second, that it seems cleaner.

So I guess I would counter-argument that it's the responsibility of OrderStatusListViewController to know which properties of the view model it needs to know about, instead of the class creating OrderStatusListViewController 😊

Reply:
Sorry, it seems like I left this in a background thread, and I just realised there is another reason. The way I see it, the OrderStatusListViewController is literally a view of a specific slice of the data encapsulated in an Order. 

Other views that only render specific slices of an Order might be the OrderDetails screen, or Order Products, or Order Refunds...

So I guess, in a way, I visualise it like this (pardon the back-of-the-envelope diagram):

<img src="https://user-images.githubusercontent.com/2722505/66973505-c45d0000-f0ca-11e9-9e60-49d949cda50b.jpg" width="350"/>

Reply:
thanks a lot for giving many thoughts to it! yea these two points make sense, and it's another way of looking at it. I was mostly coming from passing the minimum amount of data to a class.

I merged this PR sooner earlier because there were 3 PRs with release notes merge conflicts and this was the second one 😅I can update it the next time I work on Orders (I think there is an Order List design update project)

Reply:
Oh, sorry, I didn't mean my comment as a suggestion for a change, just as a little bit of a thought exercise... You know I like to be always flexing the architectural muscle 😊